### ISSUE TITLE
Other triggers off payments

### ISSUE THREAD

**Issue by sirinath** [gratipay/gratipay.com#3668]
Weekly payments are predictable. But it would be nice to tipped based on events.

E.g. flat tip or base on % of code contribution on a release for a given project or any project or group of projects,  flat tip or base on % of code committed for a given project or any project or group of projects, flat tip or base on % of code merged to master for a given project or any project or group of projects, etc.

Basically have rules to determine allocation and when the payment is made based on events and other rules.
##

**Comment by whit537** [comment]
This sounds like a good use-case for using our API ... which is currently in a bit of a messy state. :-(
